blueman/gui/manager/ManagerMenu.py

```diff
@@ -132,85 +158,56 @@ def on_device_selected(self, List, device, tree_iter):
         else:
             self.item_device.props.sensitive = False
 
-    def on_adapter_property_changed(self, lst, adapter, kv):
-        (key, value) = kv
-        if key == "Name" or key == "Alias":
-            self.generate_adapter_menu()
-        elif key == "Discovering":
+    def on_adapter_property_changed(self, _adapter, name, value, path):
+        if name == "Name" or name == "Alias":
+            item = self.adapter_items[path][0]
+            item.set_label(value)
+        elif name == "Discovering":
             if self.Search:
                 if value:
                     self.Search.props.sensitive = False
                 else:
                     self.Search.props.sensitive = True
 
-    def generate_adapter_menu(self):
-        menu = Gtk.Menu()
+    def on_adapter_selected(self, menuitem, adapter_path):
+        if menuitem.props.active:
+            if adapter_path != self.blueman.List.Adapter.get_object_path():
+                dprint("selected", adapter_path)
+                self.blueman.Config["last-adapter"] = adapter_path_to_name(adapter_path)
+                self.blueman.List.SetAdapter(adapter_path)
 
-        sep = Gtk.SeparatorMenuItem()
-        sep.show()
-        menu.append(sep)
+    def on_adapter_added(self, _manager, adapter_path):
+        adapter = bluez.Adapter(adapter_path)
+        menu = self.item_adapter.get_submenu()
+        object_path = adapter.get_object_path()
 
-        settings = create_menuitem("_Preferences", get_icon("preferences-system", 16))
-        settings.connect("activate", lambda x: self.blueman.adapter_properties())
-        settings.show()
-        menu.append(settings)
+        item = Gtk.RadioMenuItem.new_with_label(self._adapters_group, adapter['Alias'])
```

Comment:
May have been added in some recent BlueZ version. Maybe Alias just did not exist in BlueZ 4.